coordinator/statement_executor.go

```diff
@@ -492,11 +495,22 @@ func (e *StatementExecutor) createIterators(stmt *influxql.SelectStatement, ctx
 
 	// Replace instances of "now()" with the current time, and check the resultant times.
 	nowValuer := influxql.NowValuer{Now: now}
-	stmt.Condition = influxql.Reduce(stmt.Condition, &nowValuer)
-	// Replace instances of "now()" with the current time in the dimensions.
-	for _, d := range stmt.Dimensions {
-		d.Expr = influxql.Reduce(d.Expr, &nowValuer)
+	var evalTime func(stmt *influxql.SelectStatement)
+	evalTime = func(stmt *influxql.SelectStatement) {
+		stmt.Condition = influxql.Reduce(stmt.Condition, &nowValuer)
+		// Replace instances of "now()" with the current time in the dimensions.
+		for _, d := range stmt.Dimensions {
+			d.Expr = influxql.Reduce(d.Expr, &nowValuer)
+		}
+
+		for _, source := range stmt.Sources {
+			switch source := source.(type) {
+			case *influxql.SubQuery:
+				evalTime(source.Statement)
+			}
+		}
 	}
+	evalTime(stmt)
```

Comment:
Done. I made a `Reduce` function on `SelectStatement` that would call `Reduce` on the appropriate fields. I couldn't add it as part of the case statement for `Reduce` itself though because that only accepts an `Expr` and I didn't want to change the function signature.